mlflow/lightgbm.py

```diff
@@ -284,12 +318,13 @@ def callback(env):
         for imp_type in ['split', 'gain']:
             features = model.feature_name()
             importance = model.feature_importance(importance_type=imp_type)
+            plot_importance(features, importance, imp_type)
```

Comment:
Looking at Travis tests, it seems generating the feature importance plot via matplotlib can fail if the user's environment isn't properly configured (see [this SO post](https://stackoverflow.com/questions/37604289/tkinter-tclerror-no-display-name-and-no-display-environment-variable)). Given that, we should probably try-catch this plotting logic. We can initialize a logger object at the top of our module (see analogous [mlflow.tensorflow code](https://github.com/mlflow/mlflow/blob/8447c4201ae7306d69b692f5e634c7eb59cf1462/mlflow/tensorflow.py#L47) for example) & then use it to log any plotting failures that occur:

```suggestion
      try:
            log_feature_importance_plot(features, importance, imp_type)
      except Exception as e:
            _logger.exception("Failed to log feature importance plot. LightGBM autologging will ignore the failure and continue. Exception: ")
```

Reply:
We likely also need to configure matplotlib to not use the Tkinter backend in our Travis tests, as described in the above SO post.